Kickstarter-iOS/Views/PledgeCTAContainerView.swift

```diff
@@ -98,31 +89,81 @@ final class PledgeCTAContainerView: UIView {
   }
 
   // MARK: - View model
 
   override func bindViewModel() {
     super.bindViewModel()
 
-    self.vm.outputs.buttonTitleTextColor
+    self.viewModel.outputs.buttonTitleTextColor
       .observeForUI()
       .observeValues { [weak self] textColor in
         self?.pledgeCTAButton.setTitleColor(textColor, for: .normal)
       }
 
-    self.titleAndSubtitleStackView.rac.hidden = self.vm.outputs.stackViewIsHidden
-    self.titleLabel.rac.text = self.vm.outputs.titleText
-    self.subtitleLabel.rac.text = self.vm.outputs.subtitleText
-    self.pledgeCTAButton.rac.backgroundColor = self.vm.outputs.buttonBackgroundColor
-    self.pledgeCTAButton.rac.title = self.vm.outputs.buttonTitleText
-    self.spacer.rac.hidden = self.vm.outputs.spacerIsHidden
+    self.viewModel.outputs.rootStackViewAnimateIsHidden
+      .observeValues { [weak self] isHidden in
+        let duration = isHidden ? 0.0 : 0.18
+        let alpha: CGFloat = isHidden ? 0.0 : 1.0
+        UIView.animate(withDuration: duration, animations: {
+          _ = self?.rootStackView
+            ?|> \.alpha .~ alpha
+        })
+      }
+
+    self.viewModel.outputs.activityIndicatorIsAnimating
+      .observeValues { isAnimating in
+        let announcementMessage = isAnimating ? "Loading project state" : "Loading project state complete"
+
+        UIAccessibility.post(notification: .announcement, argument: announcementMessage)
+      }
+
+    self.activityIndicator.rac.animating = self.viewModel.outputs.activityIndicatorIsAnimating
+    self.titleAndSubtitleStackView.rac.hidden = self.viewModel.outputs.stackViewIsHidden
+    self.titleLabel.rac.text = self.viewModel.outputs.titleText
+    self.subtitleLabel.rac.text = self.viewModel.outputs.subtitleText
+    self.pledgeCTAButton.rac.backgroundColor = self.viewModel.outputs.buttonBackgroundColor
+    self.pledgeCTAButton.rac.title = self.viewModel.outputs.buttonTitleText
+    self.spacer.rac.hidden = self.viewModel.outputs.spacerIsHidden
   }
 
   // MARK: - Configuration
 
-  func configureWith(project: Project) {
-    self.vm.inputs.configureWith(project: project)
+  func configureWith(value: (project: Project, isLoading: Bool)) {
+    self.viewModel.inputs.configureWith(value: value)
+  }
+
+  // MARK: Functions
+
+  private func configureSubviews() {
+    _ = (self.rootStackView, self)
+      |> ksr_addSubviewToParent()
+      |> ksr_constrainViewToEdgesInParent()
+
+    _ = (self.activityIndicator, self)
+      |> ksr_addSubviewToParent()
+      |> ksr_constrainViewToCenterInParent()
+
+    _ = ([self.titleLabel, self.subtitleLabel], self.titleAndSubtitleStackView)
+      |> ksr_addArrangedSubviewsToStackView()
+
+    _ = ([self.titleAndSubtitleStackView, self.spacer, self.pledgeCTAButton], self.rootStackView)
+      |> ksr_addArrangedSubviewsToStackView()
+  }
+
+  private func setupConstraints() {
+    NSLayoutConstraint.activate([
+      self.activityIndicator.heightAnchor.constraint(equalToConstant: 30),
```

Comment:
Anyone have a suggestion for where to put these constants? They're pretty one-off.

Reply:
I think a local variable would be fine just so that we can update one value and it's clear that they're linked. Actually... why not rather constrain the width to the height and then just set one of those constants?